### ISSUE TITLE
Clickhandlers acting erratically with 1.1 if `draggable` is true

### ISSUE THREAD

**Issue by lozandier** [metafizzy/flickity#238]
With Flickity 1.1, there seems to be odd behavior if you attempt to click with `draggable` set to true. As the following youtube video demonstrates by former uses of Flickity that didn't have these issues prior to 1.1:
https://www.youtube.com/watch?v=FImIbHwN9mA&feature=youtu.be (Mobile Context)
<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/zngvEoJwv1I" frameborder="0" allowfullscreen></iframe>
https://www.youtube.com/watch?v=zngvEoJwv1I&feature=youtu.be (Random Size Context)
<iframe width="560" height="315" src="https://www.youtube.com/embed/zngvEoJwv1I" frameborder="0" allowfullscreen></iframe>
The erratic behavior is associated with items at the every ends of the visible portion of cells; essentially when the cells aren't  completely visible in the viewport ( a quarter or half of each cell isn't visible).

Even in a single column cell arrangement (see mobile-context vid), if the cell isn't completely visible in the viewport & has clickable items within the cells, click behavior similar to `#` links or `return false` happens.
 
These regressions cease to exist when `draggable` is set to false. All functions were turned off associated with events & also audited to ensure they weren't returning false or explicitly manipulating the propagation of events or the default action of events. 

Various events were investigated without any of them reproducing the odd click behavior until `draggable` was set to false

```js
  // TODO: Scroller fascade pending
  /* Initialize Scroller for Days. Sets up the configuration object for
   * scrollerjs while also initializing if necessary specific objects to extend
   * the config object that scrollerjs expectes
   * @param {string} scrollerSelector The intended DOM string associated with the intended container for a successful initialization of a scroller
   */
  initializeScroller( scrollerSelector ){

    console.log( `initializeScroller called` );

    console.info( this.picker.querySelector( scrollerSelector ) );

    const FlickityOptions = {
      cellSelector: `${this.dayDOMSelectors.item}`,
      accessiblity: true,
      cellAlign: `left`,
      draggable: false,
      contain: true,
      freeScroll: false,
      wrapAround: false,
      prevNextButtons: false,
      pageDots: false,
    };

    window.addEventListener( `load`, ()=> {

      // Colleciton Wrapper should be merged in.
      this.scroller = new Flickity( `${scrollerSelector}-collection-wrapper`, FlickityOptions );
      this.scroller.reloadCells();
      console.log( `This should've been created` );

      // ES6 Arrow function use within this function regarding events ensures that any scroller that attempts to change this
      // to be something else other than the object is ignored; most scroller
      // attempt to use `apply` to change this for events they perhaps thought
      // users would only user behavior  within a scroller's properties & methods.
      // Most common w/ scrollers that accomodate jQuery or are only for jQuery.

      // Use optimized resize to handle resizing; will have to probably make it
      // a global since the Paralla home page component can also benefit from
      // this.

      // Optimized resize event handler using RAF: https://developer.mozilla.org/en-US/docs/Web/Events/resize
      ( function() {

        const throttle = function( type, name, obj=window ) {

          let running = false;

          const func = function() {

            if ( running ) {

              return;

            }

            running = true;
            requestAnimationFrame( function() {

              obj.dispatchEvent( new CustomEvent( name ) );
              running = false;

            } );

          };

          obj.addEventListener( type, func );

        };

        /* init - you can init any event */
        throttle( `resize`, `optimizedResize` );

      } )();

      window.addEventListener( `optimizedResize`, ()=>{

        this.scroller.reloadCells();

      } );

      this.scroller.on( `settle`, () => {

        // this.verifyVisibilityOfDays();
        // this.verifyStateOfNavButtons();

      } );

      this.scroller.on( `pointerUp`, (e) => {

       // this.verifyVisibilityOfDays();
       // this.verifyStateOfNavButtons();

      } );

      this.scroller.on( `staticClick`, (e) => {

        // this.verifyVisibilityOfDay();
       //  this.verifyStateOfNavButtons();

        });


        // this.verifyVisibilityOfDays();
        // this.verifyStateOfNavButtons();

      } );
      this.scroller.on( `cellSelect`, () => {

        // this.verifyVisibilityOfDays();
        // this.verifyStateOfNavButtons();
        //
      } );
      // Make the UI more responsive by changing state of things like buttions  as dragging & pointer moves occur; Also useful if designers want to add transitions to things that aren't visible or are as user drags or moves with a pointer
      this.scroller.on( `dragMove`, () => {

        //this.verifyVisibilityOfDays();
        //this.verifyStateOfNavButtons();

      } );

      this.scroller.on( `pointerMove`, () => {

        //this.verifyVisibilityOfDays();
        //this.verifyStateOfNavButtons();

      } );

  },

```

The erratic behavior seems isolated to `draggable` being true since none of the associated funcs desired to be invoked leave possibility of returning or invoking values that change the event lifecycle; for example:

```js
  verifyVisibilityOfDays() {
    // days was a ES6 Set or Map; loops reverted to a for...of instead of a Rx.Observable or forEach for the sake of simplicity
    for( const day of this.days.values() ) {

      this.verifyVisibilityOfDay( day );

    }

  },

  verifyVisibilityOfDay( day ) {

    if ( this.isElementCompletelyVisible( day ) ) {

      if ( day.classList.contains( `not-visible` ) ) {

        day.classList.remove( `not-visible` );

      }

      day.classList.add( `visible` );

    } else {

      if ( day.classList.contains( `visible` ) ) {

        day.classList.remove( `visible` );

      }

      day.classList.add( `not-visible` );

    }

  },

```

In conclusion, it seems something seems have gone wrong with the behavior of `draggable` being true from the recent update 15 days ago;  I would love to be enlightened of a quirk

**Comment by desandro** [comment]
Thanks for reporting this issue. I appreciate you recording the movies, but a **reduced test case** would be the best way to identify the issue. See [Submitting Issues in the contributing guidelines](https://github.com/metafizzy/flickity/blob/master/contributing.md#submitting-issues).

It sounds like this bug could be caused by focusing the carousel. Can you try setting `accessibility: false` and seeing if this resolves it?

**Comment by lozandier** [comment]
@desandro Howdy, setting `accessibility: false` does *not* resolve the issue. I'll try my best to create a reduced test-case this weekend/early-next week. 

I'll also try to manually manipulate `focus`/retargeting with DOM focus/retargeting methods in the meantime as well since the difference between pointer-events & other events in how they retarget *may* be an issue here.  

Labels are supposed to be clicked that then trigger the `click` event of the inputs they represent that are visually hidden.

The click handler on the actual radio inputs (represented individually as `timeslot` in the code) is the following:

```js
        Rx.DOM.fromEvent( timeslot, `click` ).subscribe( ( e )=>{

          if ( e.target.checked === false) {

            this.rsvpBlock.textContent = this.promptToPickASlot;

          } else {

            const labelForDay = this.slotLabels.get( e.target.id );
            console.info( `labelForDay addClass`, labelForDay );

            const event = new CustomEvent( `timeslot-engagement`, {
              detail: {

                day,

              },
              bubbles: true,
              canceable: false,

            } );

            event.detail.labelForTimeslot = labelForDay;
            event.detail.timeslot = timeslot;
            e.target.dispatchEvent( event );

          }

        } );

      }








```
